### DISCUSSION TITLE
Time to update "Delete files and folders" documentation?

### DISCUSSION THREAD

**Discussion by SamuelMiller** [gulpjs/gulp#2681] [Ideas]
IMHO, it's time to update the gulp "Delete Files and Folders documentation" to either replace the vinyl path plugin with an alternative, or provide an example of how to load it as an ESM within a commonJS gulpfile.js, or of how to use it with the newish fix-esm plugin ((https://git.cryto.net/janwirth/fix-esm) to simplify the process. Vinyl path is now an ESM module making it a lot more complex to use in a commonJS gulpfile.js project--beyond my know-how.

In the good old days, one could do something similar to the following:

```
const gulp =
    { src, dest, watch, series, parallel } = require('gulp'),
    $ = require('gulp-load-plugins')({ camelize: true, lazy: true }),
    del = require('del'),
    vinylPaths = require('vinyl-paths')
};

function convertCase() {
    return src(['./src/assets/fonts/playtime/*.ttf'])
        .pipe(vinylPaths(del))
        .pipe($.rename(function (path) {
            path.dirname = path.dirname.toLowerCase(); path.basename = path.basename.toLowerCase(); path.extname = path.extname.toLowerCase();
        }))
        .pipe(dest('./src/assets/fonts'))
};
```

But with the more recent vinyl-path update, running my above function produces the dreaded "ERR_REQUIRE_ESM".

I tried playing with loading an ESM module within gulpfile.js, but its syntax is too complex for me as a novice. 

Instead, I've gone back to the depreciated, but much appreciated, "gulp-clean", for now. Although I would like to see how to use vinyl-pulgin with the newer fix-esm plugin, there is not enough documentation for me to understand to do so--it somehow involves Babel.

 Here is my current working code going back to gulp-clean:

```
function convertCase() {
    return src(['./src/assets/fonts/**/*[A-Z]*'], { base: "./" })
        .pipe($.clean({ force: true }))
        .pipe($.rename(function (path) {
            path.dirname = path.dirname.toLowerCase(); path.basename = path.basename.toLowerCase(); path.extname = path.extname.toLowerCase(); 
        .pipe(dest('.'));
};
```

(1) match filenames with any capital letters, (2) delete them, (3) rename them to lowercase, and (4) pipe the result into the same directory

**Comment by martin-badin** [comment]
I have a similar problem related to ESM modules. I am trying to upgrade `del` package but I am unsuccessful. I got this error message
`[ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for ....`. I think the issue is related to ESM modules because I didn't get the error message before upgrading the package.
